src/node_process_methods.cc

```diff
@@ -130,6 +130,71 @@ static void CPUUsage(const FunctionCallbackInfo<Value>& args) {
   fields[1] = MICROS_PER_SEC * rusage.ru_stime.tv_sec + rusage.ru_stime.tv_usec;
 }
 
+// ThreadCPUUsage use system dependent implementation to get the current thread
+// cpu times:
+//
+// - On Unix, it uses getrusage(2) with RUSAGE_THREAD.
+// - On macOS, it uses thread_info(2) with THREAD_BASIC_INFO.
+// - On Windows, it uses GetThreadTimes(2).
+//
+// Returns those values as Float64 microseconds in the elements of the array
+// passed to the function.
+static void ThreadCPUUsage(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  // Get the double array pointer from the Float64Array argument.
+  Local<ArrayBuffer> ab = get_fields_array_buffer(args, 0, 2);
+  double* fields = static_cast<double*>(ab->Data());
+
+#ifdef __linux__
+  clockid_t clockid;
+  pthread_getcpuclockid(pthread_self(), &clockid);
+
+  struct rusage usage;
+
+  if (getrusage(RUSAGE_THREAD, &usage) == -1) {
+    return THROW_ERR_THREAD_CPU_USAGE_FAILED(env);
+  }
+
+  fields[0] = MICROS_PER_SEC * usage.ru_utime.tv_sec + usage.ru_utime.tv_usec;
+  fields[1] = MICROS_PER_SEC * usage.ru_stime.tv_sec + usage.ru_stime.tv_usec;
+#elif __APPLE__
+  mach_port_t thread = mach_thread_self();
+  mach_msg_type_number_t count = THREAD_BASIC_INFO_COUNT;
+  thread_basic_info_data_t usage;
+  kern_return_t kr =
+      thread_info(thread, THREAD_BASIC_INFO, (thread_info_t)&usage, &count);
+
+  if (kr != KERN_SUCCESS) {
+    return THROW_ERR_THREAD_CPU_USAGE_FAILED(env);
+  }
+
+  fields[0] =
+      MICROS_PER_SEC * usage.user_time.seconds + usage.user_time.microseconds;
+  fields[1] = MICROS_PER_SEC * usage.system_time.seconds +
+              usage.system_time.microseconds;
+#elif _WIN32
+  HANDLE thread_ = GetCurrentThread();
```

Comment:
Can this throw? I'm not familiar with the API

Reply:
Apparently not, unless I missed something.

https://learn.microsoft.com/en-us/windows/win32/api/processthreadsapi/nf-processthreadsapi-getcurrentthread